### ISSUE TITLE
shm/copy-region doesn't always copy the highlighted region

### ISSUE THREAD

**Issue by MichaelXavier** [chrisdone/structured-haskell-mode#128]
I'm not sure if I'm misunderstanding how shm/copy-region is supposed to work, but it seems like shm's copy-region frequently gets confused and copies some unexpected part of the buffer than what is the highlighted shm region. I expect it to copy what I think shm considers to be the current "node". I'm using shm `20160211.1154`

The minimal example I can show is:

```haskell
data Foo = Foo {foo :: String|}
```

I hit `]` until `foo :: String` is under the current node face. I then hit `M-w` for `shm/copy-region`, move down a line and hit `C-y` for `shm/yank` and it looks like:

```haskell
data Foo = Foo {foo :: String|}
}
```

In other words, it pastes the `}` rather than `foo :: String`. Another example, if I move point to `f|oo` so just the word "foo" is the current node and do the same, it yanks from point onward: `oo :: String}`. Is `shm/copy-region` misbehaving or am I really looking for an as-yet non-existent `shm/copy-node`?

Thanks, and thanks so much for SHM, it has improved my Haskell life greatly!

**Comment by MichaelXavier** [comment]
As an update, I noticed you actually have the functionality for this built into `shm-kill-node` so its trivial to implement this:

```elisp
(defun shm/copy-node ()
  "Copy the current node."
  (interactive)
  (shm-kill-node nil nil nil t))
```

Now I just need to come up with a reasonable keybinding for it. Is there any interest in this being in the core functionality?

**Comment by chrisdone** [comment]
`shm/copy-region` just works on the regular Emacs region. You're supposed to use e.g. `shm/mark-node` via `C-M-SPC` to set the region to the boundaries of the node. Then `shm/copy-region` will copy the selected region.

**Comment by chrisdone** [comment]
(Sometimes you copy more than one node like n do statements or import lines, in which case `shm/copy-region` will still work as expected.)

**Comment by MichaelXavier** [comment]
Ah ok. So this is totally a misunderstanding of terminology on my part. I think I had mixed up node and region in my head. It looks like `shm/copy-node` will save me a few keystrokes from having to mark node though, so I'm happy. Thanks!

**Comment by chrisdone** [comment]
👍
